fluidkeys/secretreceive.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/fluidkeys/fluidkeys/colour"
+
+	"github.com/fluidkeys/crypto/openpgp"
+	"github.com/fluidkeys/crypto/openpgp/armor"
+	"github.com/fluidkeys/fluidkeys/api"
+	"github.com/fluidkeys/fluidkeys/out"
+	"github.com/fluidkeys/fluidkeys/pgpkey"
+)
+
+func secretReceive() exitCode {
+	out.Print("\n")
+	client := api.NewClient()
+	keys, err := loadPgpKeys()
+
+	if err != nil {
+		printFailed("Couldn't load PGP keys")
+		return 1
+	}
+
+	out.Print(colour.Info("Downloading secrets...") + "\n\n")
+
+	for _, key := range keys {
+		email, err := key.Email()
+		if err != nil {
+			printFailed("Couldn't get email for key " + key.Fingerprint().String() + "\n")
+			continue
+		}
+
+		out.Print("For " + email + ":\n\n")
+		if encryptedSecrets, err := client.ListSecrets(key.Fingerprint()); err != nil {
+			printFailed(err.Error() + "\n")
+		} else {
+			if encryptedSecrets == nil {
+				printInfo("No secrets found\n")
+				continue
+			}
+			a := LoadPrivateKeyFromGnupg{}
+			if privateKey, _, err := getDecryptedPrivateKeyAndPassword(&key, a.passwordGetter); err != nil {
```

Comment:
I'm not sure how `a.passwordGetter` could ever be set here? Also `passwordGetter`'s type is `promptForPasswordInterface`. I think this might explain the segfault I experienced :)

Reply:
Doh!